### PR TITLE
reject payment accounts with duplicate names

### PR DESCRIPTION
**Pull request by napoly**
#283

### CODE REVIEW

File: core/src/main/java/bisq/core/api/CorePaymentAccountsService.java [hunk 1]
```diff
@@ -135,6 +134,7 @@ synchronized PaymentAccount createCryptoCurrencyPaymentAccount(String accountNam
                                                       String address,
                                                       boolean tradeInstant) {
         accountService.checkAccountOpen();
+        verifyAccountNameUniqueness(accountName);
```
Comment (woodser) [inline core/src/main/java/bisq/core/api/CorePaymentAccountsService.java:137]: Please rename to `verifyAccountNameUnique` for brevity.

File: core/src/main/java/bisq/core/api/CorePaymentAccountsService.java [hunk 2]
```diff
@@ -173,6 +172,12 @@ void validateFormField(PaymentAccountForm form, PaymentAccountFormField.FieldId
         paymentAccount.validateFormField(form, fieldId, value);
     }
 
+    private void verifyAccountNameUniqueness(String accountName) {
+        if (getPaymentAccounts().stream().anyMatch(e -> e.getAccountName() != null &&
+                e.getAccountName().equals(accountName)))
+            throw new IllegalArgumentException(format("account %s is already taken", accountName));
```
Comment (woodser) [inline core/src/main/java/bisq/core/api/CorePaymentAccountsService.java:178]: Please use the error message "Account '%s' is already taken" which is more suitable for the user.

File: core/src/main/java/bisq/core/api/CorePaymentAccountsService.java [hunk 3]
```diff
@@ -145,7 +145,6 @@ synchronized PaymentAccount createCryptoCurrencyPaymentAccount(String accountNam
         Optional<CryptoCurrency> cryptoCurrency = getCryptoCurrency(currencyCode.toUpperCase());
         cryptoCurrency.ifPresent(cryptoCurrencyAccount::setSingleTradeCurrency);
         user.addPaymentAccount(cryptoCurrencyAccount);
-        if (!(cryptoCurrencyAccount instanceof AssetAccount)) accountAgeWitnessService.publishMyAccountAgeWitness(cryptoCurrencyAccount.getPaymentAccountPayload()); // TODO (woodser): applies to Haveno?
```
Comment (woodser) [inline core/src/main/java/bisq/core/api/CorePaymentAccountsService.java:148]: Why is this line removed? (The TODO can be removed since the account age witness has been integrated.)

**Review by woodser — CHANGES_REQUESTED** [review]

File: core/src/main/java/bisq/core/api/CorePaymentAccountsService.java (hunk 3 above)
Comment (napoly) [reply core/src/main/java/bisq/core/api/CorePaymentAccountsService.java:148]: sorry reverted without the TODO

File: core/src/main/java/bisq/core/api/CorePaymentAccountsService.java (hunk 2 above)
Comment (napoly) [reply core/src/main/java/bisq/core/api/CorePaymentAccountsService.java]: done

File: core/src/main/java/bisq/core/api/CorePaymentAccountsService.java (hunk 1 above)
Comment (napoly) [reply core/src/main/java/bisq/core/api/CorePaymentAccountsService.java]: done

File: core/src/main/java/bisq/core/api/CorePaymentAccountsService.java (hunk 3 above)
Comment (napoly) [reply core/src/main/java/bisq/core/api/CorePaymentAccountsService.java:148]: and actually.. if (!(cryptoCurrencyAccount instanceof AssetAccount)) is always true.. ehm false

**Review by woodser — APPROVED** [review]
